app/assets/javascripts/duties.js

```diff
@@ -0,0 +1,69 @@
+var ONE_HOUR_TO_MILLISECONDS = 1000 * 60 * 60;
+var SCROLL_LEFT_PX = 150;
+var MEDIUM_SCREEN_SIZE = 768;
+
+function setDutyTableButtons() {
+  if ($(window).width() <= MEDIUM_SCREEN_SIZE) {
+    $('.duty-table-button').removeClass('float-sm-right');
+    $('.duty-table-title').removeClass('float-left');
+  }
+}
+
+function toggleSidebar() {
+  $('#announcement-toggle-btn').on('click', function() {
+    if ($('#announcement-sidebar').css('display') == "block") {
+      $('#announcement-sidebar').hide().removeClass('open');
+
+      $('#duty-table').addClass('col-md-12');
+
+      if ( $(window).width() <= MEDIUM_SCREEN_SIZE) {
+        $('.duty-table-button').addClass('float-sm-right');
+        $('.duty-table-title').addClass('float-left');
+      }
+    } else {
+      $('#announcement-sidebar').fadeIn({
+        duration: 200
+      }).addClass('open');
+
+      $('#duty-table').addClass('col-md-9').removeClass('col-md-12');
+
+      if ( $(window).width() <= MEDIUM_SCREEN_SIZE) {
+        $('.duty-table-button').removeClass('float-sm-right');
+        $('.duty-table-title').removeClass('float-left');
+      }
+    }
+  });
+}
+
+function scrollToCurrentTime(startTime) {
+  var date = new Date();
+  var rounded = new Date(Math.floor(date.getTime() / ONE_HOUR_TO_MILLISECONDS) * ONE_HOUR_TO_MILLISECONDS);
+  var currentHour = rounded.getHours();
+  var offset = currentHour - startTime;
+
+  if (offset > 0) {
+    $('.schedule-right').scrollLeft(SCROLL_LEFT_PX * offset);
+  }
+}
+
+function colourScheduleTable(numOfPlaces) {
+  var nthChildInt = numOfPlaces * 2;
+
+  for (offset = 0; offset < numOfPlaces; offset++) {
+    $('.schedule-container tr:nth-child(' + nthChildInt + 'n -' + offset + ')').css({
+      "background-color": "#f6e1af"
```

Comment:
Would be nice to put this `f6e1af` as a constant above. Since you have been doing that anyways.